Public/Src/Engine/Scheduler/PipExecutor.cs

```diff
@@ -1404,6 +1406,25 @@ private static void MakeSharedOpaqueOutputIfNeeded(IPipExecutionEnvironment envi
 
                                 ++retryCount;
 
+                                if (result.PipProperties != null)
+                                {
+                                    if (pipProperties != null)
+                                    {
+                                        foreach (string pipPropertyKey in pipProperties.Keys)
+                                        {
+                                            result.PipProperties.TryGetValue(pipPropertyKey, out int value);
+                                            result.PipProperties[pipPropertyKey] = value + pipProperties[pipPropertyKey];

```

Comment:
It took me some time to understand what you're doing here.

I'd rather override `result.PipProperties` with `pipProperties` at the end of the loop (just like `hasRetry`). 

I.e., `pipProperties` is the dictionary used to accumulate all obtained `result.PipProperties`. Then it updates the pip properties of returned `result`.